TheaterSchedule.BLL/Services/MessageService.cs

```diff
@@ -30,30 +41,42 @@ public MessageDTO GetById(int id)
             {
                 return null;
             }
-            return new MessageDTO
-            {
-                MessageId = message.MessageId,
-                Subject = message.Subject
-            };
+
+            return _mapper.Map<MessageDTO>(message);
         }
 
-        public void SendMessage(MessageDTO newMessage)
+        public async Task SendMessage(MessageDTO newMessage)
         {
-            Account account = accountRepository.GetAccountByPhoneId(newMessage.PhoneId);
+            var account = userRepository.GetById(newMessage.AccountId);
+
             if (account == null)
             {
-                throw new ArgumentException("Non existent account");
+                throw new NullReferenceException();
             }
 
-            var message = new Entities.Models.Message()
-            {
-                Subject = newMessage.Subject,
-                MessageText = newMessage.MessageText,
-                AccountId = account.AccountId
-            };
+            var message = _mapper.Map<Message>(newMessage);
 
             messageRepository.Add(message);
-            theaterScheduleUnitOfWork.Save();
+            await theaterScheduleUnitOfWork.SaveAsync();
+        }
+
+        public List<UserMessageDTO> GetAllMessages()
+        {
+            var messages = messageRepository.GetUnrepliedMessages();
+
+            var users = userRepository.GetAll().ToList();
+
+            if (messages == null)
+            {
+                throw new ArgumentException("Not found messages");
```

Comment:
it's better just return empty list when no messages found.
Also looks like that variable `message` can't be null - it returned from repository